hw9/hw9_t03.py

```diff
@@ -0,0 +1,48 @@
+"""
+Write a function that takes directory path, a file extension and an optional tokenizer.
+
+It will count lines in all files with that extension if there are no tokenizer.
+If a the tokenizer is not none, it will count tokens.
+For dir with two files from hw1.py:
+universal_file_counter(test_dir, "txt")
+6
+universal_file_counter(test_dir, "txt", str.split)
+6
+"""
+import os
+from pathlib import Path
+from typing import Callable, Optional, Union
+
+
+def count_lines(file_path: Union[str, Path]) -> int:
+    line_counter = 0
+    with open(file_path) as file:
+        for _line in file:
+            line_counter += 1
+    return line_counter
+
+
+def count_tokens(file_path: Union[str, Path], tokenizer: Callable[[str], str]) -> int:
+    token_counter = 0
+    with open(file_path) as file:
```

Comment:
This line is repeated twice. Is it the pattern?

Reply:
Yes, this is a pattern. Do you have other patterns?